userapi/src/main/java/model/D42Model.java

```diff
@@ -0,0 +1,117 @@
+package model;
+
+
+import fk.prof.storage.AsyncStorage;
+import fk.prof.storage.S3AsyncStorage;
+import org.apache.commons.codec.binary.Base32;
+import org.slf4j.LoggerFactory;
+
+import java.time.ZonedDateTime;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+/**
+ * Interacts with the AsyncStorage based on invocations from controller
+ * Created by rohit.patiyal on 19/01/17.
+ */
+public class D42Model implements IDataModel {
+    private static final String ACCESS_KEY = "66ZX9WC7ZRO6S5BSO8TG";
+    private static final String SECRET_KEY = "fGEJrdiSWNJlsZTiIiTPpUntDm0wCRV4tYbwu2M+";
+    private static final String S3_BACKUP_ELB_END_POINT = "http://10.47.2.3:80";
+    private static final String VERSION = "v001";
+    private static final String DELIMITER = "_";
+    private static final String BUCKET = "bck1";
+    private static final String PATH_SEPARATOR = "/";
+    private static final int TIMEOUT = 2;
+    private static org.slf4j.Logger LOGGER = LoggerFactory.getLogger(D42Model.class);
+    private S3AsyncStorage store = null;
+    private Base32 base32 = null;
+
+    public D42Model() {
+        base32 = new Base32();
+        setStorage(new S3AsyncStorage(S3_BACKUP_ELB_END_POINT, ACCESS_KEY, SECRET_KEY, Executors.newCachedThreadPool()));
```

Comment:
get the storage instance as part of the constructor. The creation of the storage instance is not this class's responsibility. Storage should be initialized by the application in its startup/initialization phase.